File: dtos/FluidReadingCreationOptions.cs
using System;
using System.ComponentModel.DataAnnotations;
using HealthGyro.Models.Enums;

namespace HealthGyro.Models.Dtos
{
   public class FluidReadingCreationOptionsDto
   {
      public DateTime? RecordedAt { get; set; }
      [Required]
      public FluidReadingType Type { get; set; }
      [Required]
      public FluidReadingRoute Route { get; set; }
      [Required]
      public FluidReadingUnit Unit { get; set; }
      [Required]
      public decimal Quantity { get; set; }
   }
}

File: schemas/validation/fluid-reading-creation-options.ts
import {getRequiredValidationMessage} from './utils';
import {number, object, string} from 'yup';

const fluidReadingCreationOptions = object({
    type: string().required(getRequiredValidationMessage),
    route: string().required(getRequiredValidationMessage),
    unit: string().required(getRequiredValidationMessage),
    quantity: number().required(getRequiredValidationMessage),
});

export default fluidReadingCreationOptions;